cfn-resources/cluster/cmd/resource/resource.go

```diff
@@ -369,20 +365,45 @@ func clusterCallback(client *mongodbatlas.Client, currentModel *Model, projectID
 	if err != nil {
 		return progressEvent, nil
 	}
+
 	if progressEvent.Message == constants.Complete {
+		if !currentModel.HasAdvanceSettings() {
+			return handler.ProgressEvent{
+				OperationStatus: handler.Success,
+				Message:         "Create Success",
+				ResourceModel:   currentModel}, nil
+		}
+
 		_, _ = log.Debugf("Cluster Creation completed:%s", *currentModel.Name)
 
 		cluster, res, err := client.AdvancedClusters.Get(context.Background(), projectID, *currentModel.Name)
 		if err != nil {
 			return progress_events.GetFailedEventByResponse(fmt.Sprintf("Error creating resource : %s", err.Error()),
 				res.Response), nil
 		}
+
 		_, _ = log.Debugf("Updating cluster settings:%s", *currentModel.Name)
 		return updateClusterSettings(currentModel, client, projectID, cluster, &progressEvent)
 	}
 	return progressEvent, nil
 }
 
+func (m *Model) HasAdvanceSettings() bool {
+	/*TODO: this logic is because of a bug un Cloud Formation, when we return in_progress in the CREATE
+	,the second time the CREATE gets executed
+	it returns the AdvancedSettings is not nil but its fields are nil*/
```

Comment:
Can you create an INTMDB ticket and explain what to do to remove this TODO?

Reply:
is a super weird cfn behaviour